### PR TITLE
Bitternary bug

### PR DESCRIPTION
**Pull request by jtlap**
as a bit operation bit_ternary must return the adapted type of the second parameter (the first being the immediate op selector)

### CODE REVIEW

File: include/eve/module/core/regular/bit_ternary.hpp [hunk 1]
```diff
@@ -325,7 +332,7 @@ namespace eve
       if constexpr(K == 0xc7) return bit_or(bit_notand(c, b), bit_xor(a, bit_not(b)));
       if constexpr(K == 0xc8) return bit_and(b, bit_or(a, c));
       if constexpr(K == 0xc9) return bit_not(bit_xor(b, bit_or(a, c)));
-      if constexpr(K == 0xca) return bit_or(bit_and(a, b), bit_notand(a, c));
+      if constexpr(K == 0xca) return bit_select(a, b, c); //bit_or(bit_and(a, b), bit_notand(a, c));
```
Comment (DenisYaroshevskiy) [inline include/eve/module/core/regular/bit_ternary.hpp:335]: clang format please. The comments are not formatted.

File: include/eve/module/core/regular/bit_ternary.hpp [hunk 2]
```diff
@@ -295,19 +302,19 @@ namespace eve
...
-      if constexpr(K == 0xac) return bit_select(a, c, b);
+      if constexpr(K == 0xac) return bit_select(a, c, b);//bit_or(bit_and(c, a), bit_andnot(b, a));
       if constexpr(K == 0xad) return bit_or(bit_and(b, c), bit_xor(a, bit_not(c)));
       if constexpr(K == 0xae) return bit_or(c, bit_notand(a, b));
       if constexpr(K == 0xaf) return bit_ornot(c, a);
       if constexpr(K == 0xb0) return bit_notand(bit_notand(c, b), a);
-      if constexpr(K == 0xb1) return bit_select(c, a, bit_not(b));
-      if constexpr(K == 0xb2) return bit_select(b, bit_and(a, c), bit_or(a, c));
+      if constexpr(K == 0xb1) return bit_or(bit_and(a, c), bit_andnot(bit_not(b), c));
+      if constexpr(K == 0xb2) return bit_or(bit_and(b, a, c),bit_notand(b, bit_or(a, c)));
```
Comment (DenisYaroshevskiy) [inline include/eve/module/core/regular/bit_ternary.hpp:311]: Why are these changing? Was the previous implementation buggy? Or is it just better?

Comment (jtlap) [reply include/eve/module/core/regular/bit_ternary.hpp]: buggy

File: include/eve/module/core/regular/bit_ternary.hpp (hunk 1 above)
Comment (jtlap) [reply include/eve/module/core/regular/bit_ternary.hpp]: I want there one line by constexpr here. If you prefer I can suppress the comments.
However, what is the comment format you want here ?

File: include/eve/module/core/regular/impl/simd/x86/bit_ternary.hpp [hunk 3]
```diff
@@ -19,38 +19,82 @@
...
+        if constexpr(K == 0x53) return bit_or(bit_notand(c, a), bit_notand(a, bit_not(b)));
+        if constexpr(K == 0x72) return bit_or(bit_notand(b, c), bit_notand(c, a));
+        if constexpr(K == 0x74) return bit_or(bit_notand(c, b), bit_notand(b, a));
+        if constexpr(K == 0x8b) return bit_or(bit_and(b, c),  bit_notand(b, bit_not(a)));
+        if constexpr(K == 0x8d) return bit_or(bit_and(c, b), bit_notand(c,bit_not(a)));
+        if constexpr(K == 0xa3) return bit_or(bit_and(a, c), bit_notand(a,bit_not(b)));
+        if constexpr(K == 0xb1) return bit_or(bit_and(a, c), bit_andnot(bit_not(b), c));
+        if constexpr(K == 0xc5) return bit_or(bit_and(a, b), bit_notand(a, bit_not(c)));
+        if constexpr(K == 0xd1) return bit_or(bit_not(bit_or(b, c)), bit_and(a, b));
+        return bit_ternary.behavior(cpu_{}, opts, ik, a, b, c);
```
Comment (jfalcou) [inline include/eve/module/core/regular/impl/simd/x86/bit_ternary.hpp:54]: missing else

File: test/unit/module/core/bit_ternary.cpp [hunk 4]
```diff
@@ -216,8 +220,8 @@ TTS_CASE_WITH("Check behavior of bit_ternary",
   TTS_EQUAL(eve::bit_ternary(std::integral_constant<std::uint8_t, 0xae>(), a, b, c),  bit_or(c, bit_notand(a, b)));
   TTS_EQUAL(eve::bit_ternary(std::integral_constant<std::uint8_t, 0xaf>(), a, b, c),  bit_ornot(c, a));
   TTS_EQUAL(eve::bit_ternary(std::integral_constant<std::uint8_t, 0xb0>(), a, b, c),  bit_notand(bit_notand(c, b), a));
-  TTS_EQUAL(eve::bit_ternary(std::integral_constant<std::uint8_t, 0xb1>(), a, b, c),  bit_select(c, a, bit_not(b)));
-  TTS_EQUAL(eve::bit_ternary(std::integral_constant<std::uint8_t, 0xb2>(), a, b, c),  bit_select(b, bit_and(a, c), bit_or(a, c)));
+  TTS_EQUAL(eve::bit_ternary(std::integral_constant<std::uint8_t, 0xb1>(), a, b, c),    bit_or(bit_and(a, c), bit_notand(c, bit_not(b))));//  bit_select(c, a, bit_not(b)));
```
Comment (jfalcou) [inline test/unit/module/core/bit_ternary.cpp:223]: remove the comments

**Review by jfalcou — CHANGES_REQUESTED** [review]

File: test/unit/module/core/bit_ternary.cpp (hunk 4 above)
Comment (jtlap) [reply test/unit/module/core/bit_ternary.cpp]: done

**Review by jfalcou — APPROVED** [review]

**Pull request merged**